orchestrator/gravity_utils/src/connection_prep.rs

```diff
@@ -269,68 +268,90 @@ pub async fn check_delegate_addresses(
                     delegate_orchestrator_address, req_delegate_orchestrator_address
                 );
                 error!("In order to resolve this issue you should double check your input value or re-register your delegate keys");
-                exit(1);
+                Err(ValidityCheckError::EthreumOrchestratorAddressesIncorrect {
+                    delegate_eth_address,
+                    req_delegate_eth_address,
+                    delegate_orchestrator_address,
+                    req_delegate_orchestrator_address,
+                })
             } else if req_delegate_eth_address != delegate_eth_address {
                 error!("Your Delegate Ethereum address is incorrect!");
                 error!(
                     "You provided {}  Correct Value {}",
                     delegate_eth_address, req_delegate_eth_address
                 );
                 error!("In order to resolve this issue you should double check how you input your eth private key");
-                exit(1);
+                Err(ValidityCheckError::EthereumAddressIncorrect {
+                    delegate_eth_address,
+                    req_delegate_eth_address,
+                })
             } else if req_delegate_orchestrator_address != delegate_orchestrator_address {
                 error!("Your Delegate Orchestrator address is incorrect!");
                 error!(
                     "You provided {}  Correct Value {}",
-                    delegate_eth_address, req_delegate_eth_address
```

Comment:
Also I think this is a bug? The addresses in question are those from orchestrator and not eth? and in error! eth addresses are provided?